modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByScrollRequest.java

```diff
@@ -422,11 +423,6 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeFloat(requestsPerSecond);
         if (out.getVersion().onOrAfter(BulkByScrollTask.V_5_1_0_UNRELEASED)) {
             out.writeVInt(slices);
-        } else {
-            if (slices > 1) {
-                throw new UnsupportedOperationException("Attempting to send sliced reindex-style request to a node that doesn't support "
```

Comment:
Now that I think about this I think a better alternative is to switch this to `IllegalArgumentException` and to catch that in `assertVersionSerializable`. Then we would:
1. Fail to process requests with slices if sending them to a node that doesn't support them.
2. Return a 400 level error if this happens to happen over REST (unlikely, I think. maybe impossible)
3. Have a way for other requests to solve this problem in ways that are less surprising than attempting to degenerate into another request.